checkstyle/suppressions.xml

```diff
@@ -151,6 +151,10 @@
     <suppress checks="NPathComplexity"
               files="(ProcessorStateManager|InternalTopologyBuilder|StreamsPartitionAssignor|StreamThread).java"/>
 
+    <!-- suppress FinalLocalVariable outside of the streams package. -->
+    <suppress checks="FinalLocalVariable"
+              files="^(?!.*[\\/]streams[\\/].*$)"/>
+
```

Comment:
Making sure it only applies to Streams

Reply:
Should we make this more restrict? We don't know if the might be any other (sub)package being added with this name. Maybe better check prefix: `org.apache.kafka.streams.*` ?

Reply:
+1 on the prefix package name.